variables.tf

```diff
@@ -803,3 +803,59 @@ variable "strict_subscription_association" {
   description = "If set to true, subscriptions associated to management groups will be exclusively set by the module and any added by another process will be removed. If set to false, the module will will only enforce association of the specified subscriptions and those added to management groups by other processes will not be removed."
   default     = true
 }
+
+variable "policy_compliance_message_not_supported_definitions" {
```

Comment:
Just a thought, but would it be possible to remove this and replace with conditional logic around when a compliance message is applied to an assignment by dynamically checking the `mode` value?

Reply:
@krowlandson I looked into this, but I can't find a way to retrieve the `mode` value. The data source for policy does not include it, so no simple way to get it for built in policies. Hence this hacky workaround.

Reply:
Quick update on this. The reason this cannot be achieved is that the `azurerm_policy_definition` data source is missing the `Mode` property. I am in the process of submitting a PR to the provider to add this property. I am not sure how long it will take to make it into a provider release. If it looks like it will take a while I will ask for the PR to be approved as is and then open a separate PR to update this.

Reply:
Here is the PR on the azurerm provider: https://github.com/hashicorp/terraform-provider-azurerm/pull/20420